src/zigbee2mqtt/zigbee2mqtt-device.ts

```diff
@@ -0,0 +1,261 @@
+          } else {
+            const isWriteOnly = (expose.access ?? 0) == WRITE_BIT;
+
+            if (isWriteOnly) {
+              this.createAction(expose);
+            } else {
+              this.createProperty(expose);
+            }
+          }
+          break;
+      }
+    }
+  }
+
+  private createLightProperties(expose: Expos): void {
+    if (expose.features) {
+      ((this as unknown) as { '@type': string[] })['@type'].push('Light');
+
+      for (const feature of expose.features) {
+        if (feature.name) {
+          switch (feature.name) {
+            case 'state':
+              {
+                console.log(`Creating property for ${feature.name}`);
+
+                const property = new OnOffProperty(
+                  this,
+                  feature.name,
+                  feature,
+                  this.client,
+                  this.deviceTopic
+                );
+
+                this.addProperty(property);
+              }
+              break;
+            case 'brightness':
+              {
+                console.log(`Creating property for ${feature.name}`);
+
+                const property = new BrightnessProperty(
+                  this,
+                  feature.name,
+                  feature,
+                  this.client,
+                  this.deviceTopic
+                );
+
+                this.addProperty(property);
+              }
+              break;
+            case 'color_temp':
+              {
+                console.log(`Creating property for ${feature.name}`);
+
+                const property = new ColorTemperatureProperty(
+                  this,
+                  feature.name,
+                  feature,
+                  this.client,
+                  this.deviceTopic
+                );
+
+                this.addProperty(property);
+              }
+              break;
+            case 'color_xy':
+              {
+                console.log(`Creating property for ${feature.name}`);
+
+                const property = new ColorProperty(
+                  this,
+                  'color',
+                  feature,
+                  this.client,
+                  this.deviceTopic
+                );
+
+                this.addProperty(property);
+              }
+              break;
+          }
+        } else {
+          console.log(`Ignoring property without name: ${JSON.stringify(expose, null, 0)}`);
+        }
+      }
+    } else {
+      console.warn(`Expected features array in light expose: ${JSON.stringify(expose)}`);
+    }
+  }
+
+  private createEvents(values: string[]): void {
+    if (Array.isArray(values)) {
+      if (values.length > 0) {
+        for (const value of values) {
+          console.log(`Creating property for ${value}`);
+
+          this.addEvent(value, {
+            name: value,
+          });
+        }
+      } else {
+        console.log(`Expected object but got ${typeof values}`);
```

Comment:
Log is wrong here. It's just an empty array.

Reply:
done